jobs/jobs/airflow_utils.py

```diff
@@ -46,6 +56,76 @@ async def get_dags() -> List[pipeline.AnyPipeline]:
         return dag_api.get_dags()
 
 
+async def get_dag_status(
+    dag_id: str, job_id: int, api_client: client.ApiClient
+) -> str:
+    """Fetches the DAG run status for the given DAG ID."""
+
+    # api_client = airflow_connection()
+
+    try:
+        dag_run_id = f"{dag_id}:{job_id}"
+        api_instance = DAGRunApi(api_client)
+
+        api_response = api_instance.get_dag_run(dag_id, dag_run_id)
+
+        dag_state = api_response.state.value.lower()
+        if dag_state == AirflowPiplineStatus.success.value:
+            return AirflowPiplineStatus.success.value
+        elif dag_state == AirflowPiplineStatus.failed.value:
+            return AirflowPiplineStatus.failed.value
+        else:
+            # if other states like "queued", "running", etc.
+            raise RuntimeError(f"Unexpected DAG run state: {dag_state}")
+
+    except ApiException as e:
+        raise RuntimeError(f"Error fetching DAG run status: {e}") from e
+    except Exception as e:
+        raise RuntimeError(f"Unexpected error occurred: {e}") from e
+
+
+async def activate_dag(dag_id: str, api_client: client.ApiClient) -> None:
+    try:
+        api_instance = DAGApi(api_client)
+
+        dag = api_instance.get_dag(dag_id)
+        if dag.is_paused:
+            updated_dag = DAG(is_paused=False)
+            api_instance.patch_dag(dag_id, updated_dag)
+    except ApiException as e:
+        raise RuntimeError(f"Failed to fetch or update DAG state: {e}") from e
```

Comment:
no need to add caught exception to a new one, we have a traceback for that. Unless you want to put specific info there, then do that